core/templates/dev/head/pages/questions_list/QuestionsListDirective.ts

```diff
@@ -180,20 +204,24 @@ oppia.directive('questionsList', [
               ]
             });
 
-            modalInstance.result.then(function(skillId) {
-              $scope.skillId = skillId;
-              EditableSkillBackendApiService.fetchSkill(
-                skillId).then(
-                function(skillDict) {
-                  $scope.misconceptions = skillDict.misconceptions.map(function(
-                      misconceptionsBackendDict) {
-                    return MisconceptionObjectFactory.createFromBackendDict(
-                      misconceptionsBackendDict);
+            modalInstance.result.then(function(skillIds) {
+              $scope.skillIds = skillIds;
+              for (var i = 0; i < skillIds.length; i++) {
```

Comment:
Instead of calling EditableSkillBackendApiService.fetchSkill multiple times in a loop, I would suggest to create a new function in EditableSkillBackendApiService that can fetch 'n' number of skills in one call (It can return a list of skills, keyed by skill id maybe?)

Reply:
Done.